src/test/java/core/basesyntax/HelloWorldTest.java

```diff
@@ -1,8 +1,145 @@
 package core.basesyntax;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.exceptions.RegistrationFailureException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
 /**
  * Feel free to remove this class and create your own.
  */
 public class HelloWorldTest {
+    private static RegistrationService registrationService;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void register_addUser_Ok() {
+        User user = new User();
+        user.setLogin("Marco123");
+        user.setPassword("12345678");
+        user.setAge(20);
+
+        User result = registrationService.register(user);
+        User storageUser = registrationService.getUser(user.getLogin());
+        assertEquals(storageUser, result);
+    }
+
+    @Test
+    void register_passwordEdges_Ok() {
+        User user1 = new User();
+        user1.setLogin("Marco1");
+        user1.setPassword("123456");
+        user1.setAge(27);
+
+        User result = registrationService.register(user1);
+        assertEquals(result, user1);
+
+        User user2 = new User();
+        user2.setLogin("Parmezan");
+        user2.setPassword("12345678910");
+        user2.setAge(50);
+        result = registrationService.register(user2);
+        assertEquals(result, user2);
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        User user = null;
+        assertThrows(RegistrationFailureException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_addIdenticalUsers_notOk() {
+        User user = new User();
+        user.setLogin("Antonio");
+        user.setPassword("12345678");
+        user.setAge(23);
+
+        registrationService.register(user);
```

Comment:
this part of inserting the first user, is an arrangement of the test data, which should not be done at least with the same method that is under the test, because we are not sure that it is working as intended at the moment. So we should use some other, more low-level approaches here.
check common mistakes